src/components/TextBoxSearch.ts

```diff
@@ -0,0 +1,73 @@
+import { ChangeEvent, Component, createElement } from "react";
+
+import "./ui/TextBoxSearch.scss";
+
+export interface TextBoxSearchProps {
+    defaultQuery: string;
+    placeholder?: string;
+    onTextChangeAction: (query: string) => void;
+}
+
+interface TextBoxSearchState {
+    query: string;
+}
+
+export class TextBoxSearch extends Component<TextBoxSearchProps, TextBoxSearchState> {
+
+    constructor(props: TextBoxSearchProps) {
+        super(props);
+
+        this.state = { query: "" };
+        this.resetQuery = this.resetQuery.bind(this);
+        this.handleOnChange = this.handleOnChange.bind(this);
+    }
+
+    render() {
+        return createElement("div", { className: "search-bar" },
+            createElement("input", {
+                className: "form-control",
+                onChange: this.handleOnChange,
+                placeholder: this.props.placeholder,
+                value: this.state.query
+            }),
+            createElement("button",
```

Comment:
Be consistent. Either the brackets. see `createElement("input", {`  and `    ` createElement("button",` Decide on one.

Reply:
Thanx

Reply:
Fixed